experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/ResourceManagerCallbackHandler.scala

```diff
@@ -60,19 +76,19 @@ class ResourceManagerCallbackHandler(appConfig: AppConfig, am: ActorRef) extends
 
     //todo:wont work with non uniform containers
     (0 until requestCount).foreach(request => {
-      am ! ContainerRequestMessage(containerCount, appConfig.getEnv(GEARPUMPMASTER_VCORES).toInt)
+      resourceManagerClient ! AdditionalContainersRequest(containerCount)
```

Comment:
ResourceManagerCallbackHandler.getProgress is inherited from AMRMClientAsync.CallbackHandler and prevents moving container counts into YarnApplicationMaster.